test/test_torch.py

```diff
@@ -1887,6 +1887,32 @@ def _test_gather(self, cast, test_bounds=True):
         def test_gather(self):
             self._test_gather(self, lambda t: t)
 
+        def test_ravel(self):
+            # Test that flatten returns 1-dim tensor when given a 0-dim tensor
+            zero_dim_tensor = torch.tensor(123)
+            flat0 = zero_dim_tensor.ravel()
+            one_dim_tensor = torch.tensor([123])
+            flat1 = zero_dim_tensor.ravel()
+
+            self.assertEqual(zero_dim_tensor.shape, torch.Size([]))
+            self.assertEqual(flat0.shape, torch.Size([1]))
+            self.assertEqual(one_dim_tensor.shape, torch.Size([1]))
+            self.assertEqual(flat1.shape, torch.Size([1]))
+            self.assertEqual(flat0, one_dim_tensor)
+            self.assertEqual(flat0, flat1)
+            self.assertEqual(flat0.shape, flat1.shape)
+
+            # Test both float tensor and quantized tensor
+            tensors = [torch.randn(5, 5, 5, 5),
+                       torch._empty_affine_quantized([5, 5, 5, 5],
+                                                     scale=2,
+                                                     zero_point=3,
+                                                     dtype=torch.quint8)]
+            for src in tensors:
+                flat = src.ravel()
+                self.assertEqual(flat.shape, torch.Size([625]))
+                self.assertEqual(src.view(-1), flat.view(-1))
```

Comment:
self.assertEqual checks equality of values. It doesn't actually check if `flat` is a view of `src` or not, which we do want to check, because that is part of the semantics for ravel()